src/extension.ts

```diff
@@ -128,46 +135,74 @@ export function getXmlUri(uri: vscode.Uri): vscode.Uri {
-  const conn = config("conn");
-  let connInfo = `${conn.host}:${conn.port}[${conn.ns}]`;
+  const workspaceFolder = currentWorkspaceFolder();
+  if (clearCookies) {
+    /// clean-up cached values
+    workspaceState.update(workspaceFolder + ":host", undefined);
+    workspaceState.update(workspaceFolder + ":port", undefined);
+    workspaceState.update(workspaceFolder + ":password", undefined);
+    workspaceState.update(workspaceFolder + ":apiVersion", undefined);
+    workspaceState.update(workspaceFolder + ":docker", undefined);
+  }
+  let api = new AtelierAPI(workspaceFolder);
+  const { active, host = "", port = 0, ns = "" } = api.config;
+  let connInfo = `${host}:${port}[${ns}]`;
+  if (!host.length || !port || !ns.length) {
+    connInfo = packageJson.displayName;
+  }
   panel.text = connInfo;
   panel.tooltip = "";
-  vscode.commands.executeCommand("setContext", "vscode-objectscript.connectActive", conn.active);
-  if (!conn.active) {
-    panel.text = `${connInfo} - Disabled`;
+  vscode.commands.executeCommand("setContext", "vscode-objectscript.connectActive", active);
+  if (!active) {
+    panel.text = `${packageJson.displayName} - Disabled`;
     return;
   }
-  workspaceState.update(currentWorkspaceFolder() + ":port", undefined);
-  const { port: dockerPort, docker: withDocker } = portFromDockerCompose();
-  workspaceState.update(currentWorkspaceFolder() + ":docker", withDocker);
-  if (withDocker) {
-    terminalWithDocker();
-    if (dockerPort !== conn.port) {
-      workspaceState.update(currentWorkspaceFolder() + ":port", dockerPort);
+  if (!workspaceState.get(workspaceFolder + ":port") && !api.externalServer) {
+    const { port: dockerPort, docker: withDocker } = portFromDockerCompose();
+    workspaceState.update(workspaceFolder + ":docker", withDocker);
+    if (withDocker) {
+      if (!dockerPort) {
+        outputChannel.appendLine(
+          `Something wrong with your docker-compose connection settings. Or your service not running.`
+        );
+        panel.text = `${packageJson.displayName} - ERROR`;
+        return;
+      }
+      terminalWithDocker();
+      if (dockerPort !== port) {
+        workspaceState.update(workspaceFolder + ":host", "localhost");
+        workspaceState.update(workspaceFolder + ":port", dockerPort);
+      }
+      connInfo = `localhost:${dockerPort}[${ns}]`;
     }
-    connInfo = `${conn.host}:${dockerPort}[${conn.ns}]`;
   }
 
-  const api = new AtelierAPI(currentWorkspaceFolder());
   if (clearCookies) {
     api.clearCookies();
+  } else if (connectionSocket && connectionSocket.url == api.xdebugUrl() && connectionSocket.OPEN) {
+    panel.text = `${connInfo} - Connected`;
+    return;
   }
-  if (connectionSocket && connectionSocket.url == api.xdebugUrl() && connectionSocket.OPEN) {
+  api = new AtelierAPI(workspaceFolder);
+  if (!api.config.host || !api.config.port || !api.config.ns) {
+    outputChannel.appendLine("host, port and ns should be specified");
```

Comment:
```suggestion
    outputChannel.appendLine("host, port and ns must be specified.");
```

Reply:
Also just noticed that you haven't yet added the dependency on `intersystems-community.servermanager` into `package.json`.